src/scalib_ext/scalib/src/sasca/factor_graph.rs

```diff
@@ -309,6 +309,9 @@ impl FactorGraph {
                                 PublicValue::Single(cv) if vars_neg[0] => {
                                     PublicValue::Single(self.not(cv))
                                 }
+                                PublicValue::Multi(cv) if vars_neg[0] => {
+                                    PublicValue::Multi(cv.iter().map(|v| self.not(*v)).collect())
+                                }
```

Comment:
That would be cleaner by creating a `PublicValue::not` method, but this works too.